src/platforms/common/swdptap.c

```diff
@@ -140,7 +141,7 @@ static bool swdptap_seq_in_parity(uint32_t *ret, size_t clock_cycles)
 	*ret = result;
 	/* Terminate the read cycle now */
 	swdptap_turnaround(SWDIO_STATUS_DRIVE);
-	return parity & 1U;
+	return parity != bit;
```

Comment:
Rather than using `!=`, what does the codegen look like for `^`? If memory serves, the original `+=` pattern (which, granted, should have been `^=`) produces slightly fewer instructions than the comparison, so we'd be interested to see what different operations for calculating if the parity matches or doesn't does to this

Reply:
Comparing two bools here can be done with `!=` or `^` -- both compile to `eor.w r0, r1, r2;` right before function epilogue `add sp, #16; pop {r4, r5, r6, pc}` or `ldmia.w sp!, {clobbers, pc}`.
This block is cherry-picked from PR1688 for types and const.

Reply:
Oh, perfect, in that case.. let's stick with the `!=` check as it's more obvious what it does - a better expression of intent. Glad the compiler's seeing sense and doing the good optimisation there.